### PR TITLE
Moves signal overrride warnings to it's own log file

### PR DESCRIPTION
**Pull request by ShizCalev**
I'm tired of seeing it in the runtime log. If the signals_log file exists, that means something needs to be fixed. Enjoy.

:cl: ShizCalev
admin: Moved signal overriden stack_trace warnings to it's own log file.
/:cl:

Log example:
```
[2022-09-19 23:42:22.311] movable_ztransit overridden. Use override = TRUE to suppress this warning.
 - Target: ShizCalev (/mob/dead/new_player) Proc: /datum/element/weather_listener/proc/handle_z_level_change
```
More than enough to find what you need to silence.

### CODE REVIEW

File: code/__HELPERS/logging/debug.dm [hunk 1]
```diff
@@ -62,3 +62,6 @@
 /// Logging for lua scripting
 /proc/log_lua(text)
 	WRITE_LOG(GLOB.lua_log, text)
+
+/proc/log_signal(text)
+	WRITE_LOG(GLOB.signals_log, text)
```
Comment (dragomagol) [inline code/__HELPERS/logging/debug.dm:67]: Would you mind tossing these in alphabetical order?

File: code/_globalvars/logging.dm [hunk 2]
```diff
@@ -144,3 +144,6 @@ GLOBAL_PROTECT(world_uplink_log)
 
 GLOBAL_VAR(world_virus_log)
 GLOBAL_PROTECT(world_virus_log)
+
+GLOBAL_VAR(signals_log)
+GLOBAL_PROTECT(signals_log)
```
Comment (dragomagol) [inline code/_globalvars/logging.dm:149]: Same here

File: code/game/world.dm [hunk 3]
```diff
@@ -160,6 +160,7 @@ GLOBAL_VAR(restart_counter)
 	GLOB.world_virus_log = "[GLOB.log_directory]/virus.log"
 
 	GLOB.lua_log = "[GLOB.log_directory]/lua.log"
+	GLOB.signals_log = "[GLOB.log_directory]/signals.log"
```
Comment (dragomagol) [inline code/game/world.dm:163]: And with these two

**Review by dragomagol — APPROVED** [review]
❤️

File: code/_globalvars/logging.dm [hunk 4]
```diff
@@ -21,25 +21,25 @@ GLOBAL_PROTECT(demo_log)
...
-GLOBAL_PROTECT(lastsignalers)
-
 /// Stores who uploaded laws to which silicon-based lifeform, and what the law was
 GLOBAL_LIST_EMPTY(lawchanges)
 GLOBAL_PROTECT(lawchanges)
 
+/// Keeps last 100 signals here in format: "[src] used [REF(src)] @ location [src.loc]: [freq]/[code]"
+GLOBAL_LIST_EMPTY(lastsignalers)
+GLOBAL_PROTECT(lastsignalers)
+
```
Comment (ShizCalev) [inline code/_globalvars/logging.dm:42]:
```suggestion
/// Keeps last 100 signals here in format: "[src] used [REF(src)] @ location [src.loc]: [freq]/[code]"
GLOBAL_LIST_EMPTY(lastsignalers)
GLOBAL_PROTECT(lastsignalers)

/// Stores who uploaded laws to which silicon-based lifeform, and what the law was
GLOBAL_LIST_EMPTY(lawchanges)
GLOBAL_PROTECT(lawchanges)
```

**Comment by Mothblocks** [comment]
Make sure to make a PR to https://github.com/tgstation/tgstation13.org to get this in public logs

**Pull request merged**